Add is_dir() validation for BaseFileHelper::findFiles()

zalatov opened the pull request:
| Q | A |
| --- | --- |
| Is bugfix? | yes |
| New feature? | no |
| Breaks BC? | no |
| Tests pass? | yes |
| Fixed issues |  |

Add is_dir() validation for BaseFileHelper::findFiles().
Sometimes it throws exception when file/directory is deleted or moved while executing script.

SilverFire reviewed (changes requested):
It will be good to have a test of this case.

CHANGELOG is required

samdark commented on framework/helpers/BaseFileHelper.php:
```diff
@@ -403,7 +403,7 @@ public static function findFiles($dir, $options = [])
                 if (is_file($path)) {
                     $list[] = $path;
-                } elseif (!isset($options['recursive']) || $options['recursive']) {
+                } elseif (is_dir($path) && (!isset($options['recursive']) || $options['recursive'])) {
```
Need to make sure it works well with symlinks.

samdark replied:
On Windows as well...

zalatov commented:
@SilverFire,
I don't know how test it.
We catch this error when two concurrent requests.
While first request already deleting files in dir, second request only start find files.

@samdark,
From the documentation:
`is_dir() Note that this functions follows symbolic links. It will return true if the file is actually a symlink that points to a directory.`

zalatov commented:
Added row to CHANGELOG

cebe commented:
> But... in CHANGELOG last version is 2.08, therefore this pull request is 2.0.11.

can you rebase your branch on master?

Commit:
Add is_dir() validation for BaseFileHelper::findFiles().
Sometimes it throws exception when file/directory is deleted or moved while executing script.

zalatov commented:
Pulled last changes from master

SilverFire commented:
Merged, thank you!

SilverFire commented:
https://github.com/yiisoft/yii2/commit/50953c2960549a727a9650d0db36cf21f0c542e9
